DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/SelfAssessmentCompetency.cshtml

```diff
@@ -2,160 +2,196 @@
         </h2>
-      </summary>
-      <div class="nhsuk-details__text">
-        @(Html.Raw(@Model.Competency.Description))
-      </div>
-    </details>
-  }
-  else
-  {
-    <h2 class="nhsuk-u-margin-bottom-2 nhsuk-u-font-size-24">
-      @Model.Competency.Name
-    </h2>
-    @if (Model.Competency.Description != null)
-    {
-      <div class="nhsuk-body-l">
-        @(Html.Raw(@Model.Competency.Description))
-      </div>
+        @if (Model.Competency.Description != null)
+        {
+            <div class="nhsuk-body-l">
+                @(Html.Raw(@Model.Competency.Description))
+            </div>
+        }
     }
-  }
 </div>
 <partial name="_CompetencyFlags" model="Model.Competency.CompetencyFlags" />
 <form asp-action="SelfAssessmentCompetency" asp-route-competencyNumber="@Model.CompetencyNumber" asp-route-competencyId="@Model.Competency.Id" asp-route-competencyGroupId="@Model.Competency.CompetencyGroupID">
-  @foreach (var question in @Model.Competency.AssessmentQuestions.Select((value, i) => new { i, value }))
-  {
-    if (question.value.AssessmentQuestionInputTypeID == 2)
+    @foreach (var question in @Model.Competency.AssessmentQuestions.Select((value, i) => new { i, value }))
     {
-      <partial name="SelfAssessments/_SliderQuestion" model="question.value" view-data="@(new ViewDataDictionary(ViewData) { { "index", question.i } })" />
-    }
-    else
-    {
-      <nhs-form-group nhs-validation-for="Competency.AssessmentQuestions[1].Result">
-        <partial name="SelfAssessments/_RadioQuestion" model="question.value" view-data="@(new ViewDataDictionary(ViewData) { { "index", question.i } })" />
-      </nhs-form-group>
+        if (question.value.AssessmentQuestionInputTypeID == 2)
+        {
+            <partial name="SelfAssessments/_SliderQuestion" model="question.value" view-data="@(new ViewDataDictionary(ViewData) { { "index", question.i } })" />
+        }
+        else
+        {
+            <nhs-form-group nhs-validation-for="Competency.AssessmentQuestions[1].Result">
+                <partial name="SelfAssessments/_RadioQuestion" model="question.value" view-data="@(new ViewDataDictionary(ViewData) { { "index", question.i } })" />
+            </nhs-form-group>
+        }
+        @if (question.value.SignedOff != null && question.value.SignedOff == true)
+        {
+            <div class="nhsuk-warning-callout">
+                <h3 class="nhsuk-warning-callout__label">
+                    <span role="text">
+                        <span class="nhsuk-u-visually-hidden">Important: </span>
+                        Response confirmed
+                    </span>
+                </h3>
+                <p>Your self assessment response against this question has been confirmed. Changing your response will reset confirmation.</p>
+                @if (question.value.Verified != null)
+                {
+                    string verifiedString = "";
+                    <dl class="nhsuk-summary-list">

```

Comment:
There is an identical code present in the Web\Views\Supervisor\ReviewCompetencySelfAsessment.cshtml. Can we please move that code into a partial view and make use of that partial at both places?

Reply:
Please can you raise separate tickets for 1. partial view creation and 2. incorporating the partial view at the other place. The reason I want these 2 as separate tickets is 1753 is part of an epic for improvements and that allows us to control the changes at granular level instead of one big chunk which renders it hard to test. I can cover 1st point in this ticket but we may need another one for the other functionality.

Reply:
This is addressed.